rope/contrib/autoimport/sqlite.py

```diff
@@ -67,6 +68,7 @@ def filter_package(package: Package) -> bool:
 
 
 _deprecated_default: bool = object()  # type: ignore
+thread_local = local()
```

Comment:
IIUC, creating the thread_local at the module-level like this would mean that all instances of autoimports would share the same database and database connection, even though they belong to different instances of AutoImport and more importantly that they belong to different Projects:

```python
project_a = Project("/path/to/project/a")
project_b = Project("/path/to/project/b")
ai_a = AutoImport(project_a, ...)
ai_b = AutoImport(project_b, ...)

assert ai_a.connection is not ai_b.connection
```

That does not look correct to me. I think this can be avoided by moving `thread_local` to be an instance variable initialized in `AutoImport.__init__()`.

Reply:
Thanks for pointing that our! You're right. I moved the thread local to the `__init__` and wrote a test that verifies the following passes:

```python
# Test connections
from rope.base.project import Project
from rope.contrib.autoimport.sqlite import AutoImport

ai1 = AutoImport(Project(".."))
ai2 = AutoImport(Project("."))
ai3 = AutoImport(Project("."))
assert ai1.connection is not ai2.connection
assert ai1.connection is not ai3.connection
```